crates/bevy_app/src/app.rs

```diff
@@ -631,19 +628,18 @@ impl App {
     /// App::new()
     ///     .insert_non_send_resource(MyCounter { counter: 0 });
     /// ```
-    pub fn insert_non_send_resource<T>(&mut self, resource: T) -> &mut Self
-    where
-        T: 'static,
-    {
-        self.world.insert_non_send(resource);
+    pub fn insert_non_send_resource<R: 'static>(&mut self, resource: R) -> &mut Self {
+        self.world.insert_non_send_resource(resource);
         self
     }
 
-    /// Initialize a resource in the current [`App`], if it does not exist yet
+    /// Initialize a resource with standard starting values by adding it to the [`World`]
```

Comment:
Minor suggestion to avoid the word "initialize" as it's often ambiguous whether it's separate to creation; perhaps instead:
> Create a resource with its standard starting values and insert it into the [`World`]
Or:
> Ensure a resource of the given type exists in the [`World`]. If it does not, insert a new instance of this resource with its standard starting values.